### PR TITLE
[5.5] Support for standard object (stdClass)

### PR DESCRIPTION
**Pull request by wffranco**
Support convertion of standard objects to json

### CODE REVIEW

File: src/Illuminate/Http/Response.php [hunk 1]
```diff
@@ -7,6 +7,7 @@
 use Illuminate\Contracts\Support\Jsonable;
 use Illuminate\Contracts\Support\Arrayable;
 use Illuminate\Contracts\Support\Renderable;
+use stdClass;
```
Comment (jhoff) [inline src/Illuminate/Http/Response.php]: use statements should be ordered by length.

**Review by jhoff — CHANGES_REQUESTED** [review]

**Comment by tillkruss** [comment]
Mind adding a test for this?

**Comment by taylorotwell** [comment]
Just do `return (array) $object;` from your controller.

**Comment by wffranco** [comment]
Sorry for the delay in responding.
@jhoff I did not know that rule, thanks for that.

@taylorotwell the problem is converting data to JSON, and your solution don't work in many cases, I will try to explain it fast.

Suppose you have one JSON attribute in a model that can get that structure: `{"a":"test","b": 5}`
This attribute have a getter: `return json_decode($value);`, that return a standard object (stdClass)

PHP always transform associative arrays in objects. Only empty arrays `[]` and ordered numeric arrays will be treated as arrays. Numeric arrays with missing numbers will be objects.

So I got 2 problems when I needed the full data to JSON.

- First problem: Empty array `[]` or casting empty objects `(array) $object` will return array instead of object.
- Second problem: Handling nested objects or collections, this could need a lot of mapping or mutations to get the proper results.

With that little fix they could be transformed properly.

In fact, casting `(object) $array` fix problems with some empty arrays that could be objects.

**Pull request closed without merging**